- Added, the much requested, NF_PING() feature.

bevanwsjones opened the pull request:
# Motivation
WHAT is it? Well son, it's the most requested feature this year, it's the `PING` functionality. A Macro for printing out the file and line number of where to code reached. For those who debug using a console... or as some would say... the correct way ;)
WHY it is needed? Because writing out the equivalent `cout` message takes time and uses precious brain space, also it was voted for.

 :kissing_heart:

bevanwsjones commented:
@greole We can add rank numbers if we want, but I will do this during MPI implementation.

greole commented:
> @greole We can add rank numbers if we want, but I will do this during MPI implementation.

Yeah, that would be helpful once we have mpi. But I would also like the option to add a custom message like `NF_PING("start doing this")` and `NF_PING("done doing that")`.

bevanwsjones commented:
So I changed things around a bit - I can put a message in the ping if we really want it, but you can't overload the macros apparently. So its either or.  My new proposal: I basically made it so `PING` uses the `NF_DINFO(msg)`, with `PING` as the message. I added a file and line write-out for the `DINFO`. This way if you want a 'real' message you use `NF_DINFO(msg)` and then `NF_PING` for just file and line.
